src/runtime_intrinsics.c

```diff
@@ -853,20 +853,20 @@ un_fintrinsic_withtype(fpext,fpext)
 
 // checked arithmetic
 /**
- * s_typemin =  ((typeof a)~0 << (runtime_nbits - 1))
- * s_typemax = ~((typeof a)1 << (runtime_nbits - 1))
+ * s_typemin = ((typeof a)1 << (runtime_nbits - 1))
+ * s_typemax = ((typeof a)1 << (runtime_nbits - 1)) - 1
  * u_typemin = 0
  * u_typemax = ((typeof a)1 << runtime_nbits) - 1
  * where (a - a) == (typeof(a)0
  **/
 #define sTYPEMIN(a) \
     (8 * sizeof(a) == runtime_nbits \
-     ? ((a - a + ~0) << (8 * sizeof(a) - 1)) \
-     : ((a - a + ~0) << (runtime_nbits - 1)))
+     ? -(sTYPEMAX(a) - 1) \
```

Comment:
```suggestion
     (-sTYPEMAX(a) - 1)
```

Reply:
did you mean to delete the `?`?

Reply:
Yeah, it’s not conditional, it’s just the definition. (I think)